core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -576,6 +590,13 @@ case class GlobalPSBTMap(elements: Vector[GlobalPSBTRecord]) extends PSBTMap {
       .filter(element => PSBTGlobalKeyId.fromByte(element.key.head) == key)
       .asInstanceOf[Vector[T]]
   }
+
+  def combine(other: GlobalPSBTMap): GlobalPSBTMap = {
+    val newElements: Vector[GlobalPSBTRecord] =
+      other.elements.filterNot(element =>
+        this.elements.exists(_.key == element.key))
```

Comment:
here and with all of these functions, I believe it is simpler to just call `(this.elements ++ other.elements).distinct`